intermol/gromacs/gromacs_parser.py

```diff
@@ -154,36 +147,24 @@ def canonical_angle(self, params, angle, direction='into'):
             return angle, params
         else:  # currently, no angles need to be de-canonicalized
             try:
-                a_type = self.lookup_gromacs_angles[angle.__class__]
+                a_type = self.lookup_gromacs_angle_types[angle.forcetype.__class__]
             except KeyError:
                 raise UnsupportedFunctional(angle, ENGINE)
             return a_type, params
 
-    gromacs_dihedrals = {
+    gromacs_dihedral_types = {
         # TrigDihedrals are actually used for 1, 4, and 9.  Can't use lists as keys!
-        '1': ProperPeriodicDihedral,
-        '2': ImproperHarmonicDihedral,
-        '3': RbDihedral,
-        '4': ProperPeriodicDihedral,
-        '5': FourierDihedral,
-        '9': ProperPeriodicDihedral,
-        '10': RestrictedBendingDihedral,
-        '11': BendingTorsionDihedral,
-        'Trig': TrigDihedral
+        '1': ProperPeriodicDihedralType,
+        '2': ImproperHarmonicDihedralType,
+        '3': RbDihedralType,
+        '4': ProperPeriodicDihedralType,
+        '5': FourierDihedralType,
+        '9': ProperPeriodicDihedralType,
+        '10': RestrictedBendingDihedralType,
+        '11': BendingTorsionDihedralType,
+        'Trig': TrigDihedralType
         }
-
-    # have to invert manually because of canonical conversion collapse of types.
-    lookup_gromacs_dihedrals = {
-        TrigDihedral: 'Trig',
-        ImproperHarmonicDihedral: '2',
-        RbDihedral: '3',
-        FourierDihedral: '5',
-        RestrictedBendingDihedral: '10',
-        BendingTorsionDihedral: '11'
-        }
-
-    gromacs_dihedral_types = dict(
```

Comment:
The collapse only applies to `TrigDihedralType`. This is (and if I'm not mistaken always was) handled somewhat manually in the `canonical_dihedral` function when writing out. Looking [here](https://github.com/shirtsgroup/InterMol/pull/250/files#diff-a0f7c7f4d761b8f71e80987703af240aR219) you can see the special case code for trig dihedrals.

Correct me if I'm wrong but I cannot find any other uses of the `lookup_gromacs_dihedral_types` function so perhaps this was never necessary.